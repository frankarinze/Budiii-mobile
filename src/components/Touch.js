import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';
import { gStyle } from '../constants';

const Touch = ({ accessible, activeO, onPress, style, text, textStyle }) => (
  <TouchableOpacity
    accessible={accessible}
    activeOpacity={activeO}
    onPress={onPress}
    style={style}
  >
    <Text style={textStyle}>{text}</Text>
  </TouchableOpacity>
);

Touch.defaultProps = {
  accessible: true,
  activeO: gStyle.activeO,
  style: gStyle.btn,
  text: '',
  textStyle: gStyle.btnText
};

Touch.propTypes = {
  // required
  onPress: PropTypes.func.isRequired,

  // optional
  accessible: PropTypes.bool,
  activeO: PropTypes.number,
  style: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.number,
    PropTypes.object
  ]),
  text: PropTypes.string,
  textStyle: PropTypes.object
};

export default Touch;
