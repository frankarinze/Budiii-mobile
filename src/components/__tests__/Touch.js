import React from 'react';
import renderer from 'react-test-renderer';

import Touch from '../Touch';

const tree = renderer.create(<Touch onPress={jest.fn()} />).toJSON();

describe('<Touch />', () => {
  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
