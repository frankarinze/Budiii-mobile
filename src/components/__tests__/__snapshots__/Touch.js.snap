// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Touch /> renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1d2d3a",
      "borderColor": "#1d2d3a",
      "borderRadius": 4,
      "borderWidth": 0.5,
      "height": 48,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 24,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "textAlign": "center",
      }
    }
  >
    
  </Text>
</View>
`;
